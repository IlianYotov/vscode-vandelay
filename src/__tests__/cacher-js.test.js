const path = require('path')
const {cacheJsFile, processReexports, _test} = require('../cacher-js')
const {initializeSettings} = require('../settings.js')

initializeSettings()

const {processReexportNode, getSubfileData} = _test

const filePath = path.join(__dirname, '../../test-project/src/file1.js')

test('cacheJsFile', () => {
  const data = cacheJsFile(filePath)
  expect(data).toMatchSnapshot()
})

test('processReexportNode', () => {
  const lines = [
    'export {one} from "../somewhere/else"',
    'export { one  } from \'../somewhere/else\'; ',
  ]
  lines.forEach(line => {
    const fileExports = {}
    const exportArray = []
    processReexportNode(fileExports, exportArray, line)
    expect(fileExports.reexports['../somewhere/else']).toEqual(['one'])
    expect(exportArray).toEqual(['one'])
  })

  const fileExports = {}
  const exportArray = []
  const line = 'export {three, two as two2,   one} from "../somewhere/else"'
  processReexportNode(fileExports, exportArray, line)
  expect(fileExports.reexports['../somewhere/else']).toEqual(['three', 'two', 'one'])
  expect(exportArray).toEqual(['three', 'two2', 'one'])
})

test('processReexports', () => {
  const data = {
    '/src/file1.js': {
      default: 'file1',
      named: [
        'var1',
        'var2',
        'var3',
        'fn',
        'Something',
        'varDeclaredEarlier',
        'varDeclaredEarlierWithSemiColon',
        'subdir2_file1_var1',
        'subdir2_file1_var2_renamed',
        'subdir2_file2_var1_renamed',
        'subdir2_file2_var2',
      ],
      types: [
        'MyType1',
        'MyType2',
      ],
      all: [
        './subdir1/file1',
        './subdir1/file2',
        './subdir1/fileWithNoNamedExports',
        './subdir1/nonexistentFile',
      ],
      reexports: {
        './subdir2/file1': [
          'subdir2_file1_var1',
          'subdir2_file1_var2',
        ],
        './subdir2/file2': [
          'subdir2_file2_var1',
          'subdir2_file2_var2',
        ]
      }
    },
    '/src/subdir1/file1.js': {
      named: ['subdir1_file1_var1', 'subdir1_file1_var2']
    },
    '/src/subdir1/file2.jsx': {
      named: ['subdir1_file2_var1']
    },
    '/src/subdir1/fileWithNoNamedExports.js': {
      default: 'fileWithNoNamedExports'
    },
    '/src/subdir2/file1.js': {
      reexported: ['preexisting_reexport_1', 'preexisting_reexport_2']
    },
    '/src/subdir2/file2.jsx': {},
  }

  processReexports(data)
  expect(data).toMatchSnapshot()
})

test('getSubfileData', () => {
  let data = {'/src/subdir1/file1.js': 1}
  expect(getSubfileData('/src/file1', './subdir1/file1', data)).toBe(1)
  data = {'/src/subdir1/file1.jsx': 2}
  expect(getSubfileData('/src/file1', './subdir1/file1', data)).toBe(2)
})
