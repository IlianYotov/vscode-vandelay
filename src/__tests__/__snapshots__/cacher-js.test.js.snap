// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cacheJsFile 1`] = `
Object {
  "/src/file1.js": Object {
    "all": Array [
      "./subdir1/file1",
      "./subdir1/file2",
    ],
    "default": "file1",
    "named": Array [
      "var1",
      "var2",
      "var3",
      "fn",
      "Something",
      "varDeclaredEarlier",
      "varDeclaredEarlierWithSemiColon",
      "subdir2_file1_var1",
      "subdir2_file1_var2_renamed",
      "subdir2_file2_var1_renamed",
    ],
    "reexports": Object {
      "./subdir2/file1": Array [
        "subdir2_file1_var1",
        "subdir2_file1_var2",
      ],
      "./subdir2/file2": Array [
        "subdir2_file2_var1",
      ],
    },
    "types": Array [
      "MyType1",
      "MyType2",
    ],
  },
}
`;

exports[`processReexports 1`] = `
Object {
  "/src/file1.js": Object {
    "default": "file1",
    "named": Array [
      "var1",
      "var2",
      "var3",
      "fn",
      "Something",
      "varDeclaredEarlier",
      "varDeclaredEarlierWithSemiColon",
      "subdir2_file1_var1",
      "subdir2_file1_var2_renamed",
      "subdir2_file2_var1_renamed",
      "subdir2_file2_var2",
      "subdir1_file1_var1",
      "subdir1_file1_var2",
      "subdir1_file2_var1",
    ],
    "types": Array [
      "MyType1",
      "MyType2",
    ],
  },
  "/src/subdir1/file1.js": Object {
    "named": Array [
      "subdir1_file1_var1",
      "subdir1_file1_var2",
    ],
    "reexported": Array [
      "subdir1_file1_var1",
      "subdir1_file1_var2",
    ],
  },
  "/src/subdir1/file2.jsx": Object {
    "named": Array [
      "subdir1_file2_var1",
    ],
    "reexported": Array [
      "subdir1_file2_var1",
    ],
  },
  "/src/subdir1/fileWithNoNamedExports.js": Object {
    "default": "fileWithNoNamedExports",
  },
  "/src/subdir2/file1.js": Object {
    "reexported": Array [
      "preexisting_reexport_1",
      "preexisting_reexport_2",
      "subdir2_file1_var1",
      "subdir2_file1_var2",
    ],
  },
  "/src/subdir2/file2.jsx": Object {
    "reexported": Array [
      "subdir2_file2_var1",
      "subdir2_file2_var2",
    ],
  },
}
`;
